sdk/core/azure_core/src/http/pipeline.rs

```diff
@@ -34,14 +34,17 @@ use typespec_client_core::http::{self, policies::Policy};
 /// cannot be enforced by code). All policies except Transport policy can assume there is another following policy (so
 /// `self.pipeline[0]` is always valid).
 #[derive(Debug, Clone)]
-pub struct Pipeline(http::Pipeline);
+pub struct Pipeline<F: Format = JsonFormat>(http::Pipeline<F>);
 
-impl Pipeline {
+impl<F: Format> Pipeline<F> {
```

Comment:
Interesting. Was this a necessary viral change? Though, it's congruent with our discussion offline about requiring separate clients for JSON, XML, or whatever future format we might support. @JeffreyRichter, thoughts? I can catch you up offline. Basically, the whole point of this PR is to simplify callers because the client knows what the format is, so there's no reason to pass that responsibility off to callers as we have been.

I guess what I wasn't expecting is that we couldn't do this only in client methods' impl, but is that even a practical concern? Would we have any scenarios where - using the same client instance - we want to send e.g., both JSON and XML? Seems incredibly unlikely, though not impossible. (Unbranding might be more of a concern.)

Reply:
Are you referring specifically to the trait bound on `F`? The type parameter is certainly viral though, as only the client itself knows what format to use for a `Pipeline`.

Reply:
Let's discuss offline

Reply:
I put something on our calendars, but we can chat about it as well if people would rather.